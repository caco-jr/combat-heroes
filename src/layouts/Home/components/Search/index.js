import React, { useState } from 'react';

import { SearchWrapper, SearchInput } from './index.style';

const Search = () => {
  const [value, setValue] = useState('');

  const handleChange = event => {
    setValue(event.target.value);
  };

  const handleSubmit = event => {
    event.preventDefault();
  };

  return (
    <SearchWrapper onSubmit={handleSubmit}>
      <SearchInput name="search" value={value} onChange={handleChange} />
    </SearchWrapper>
  );
};

export default Search;
