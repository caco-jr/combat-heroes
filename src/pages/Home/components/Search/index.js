import React from 'react';

const Search = () => {
  return (
    <form>
      <input name="search" />
    </form>
  );
};

export default Search;
