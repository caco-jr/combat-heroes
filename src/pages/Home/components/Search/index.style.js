import styled from 'styled-components';

export const SearchWrapper = styled.form``;

export const SearchInput = styled.input`
  border: 1px solid var(--border-color);
  border-radius: var(--border-radius);
`;
