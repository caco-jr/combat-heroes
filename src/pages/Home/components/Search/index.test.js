import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Search from './index';

describe('With Enzyme', () => {
  it('renders', () => {
    const wrapper = shallow(<Search />);

    expect(wrapper.exists()).toBe(true);
  });
});

describe('With Snapshot Testing', () => {
  it('Rendering Search', () => {
    const component = renderer.create(<Search />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
