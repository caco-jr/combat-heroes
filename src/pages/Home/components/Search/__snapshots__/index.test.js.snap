// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Rendering Search 1`] = `
<form
  className="indexstyle__SearchWrapper-sc-1r0ek5c-0 hEclBY"
>
  <input
    className="indexstyle__SearchInput-sc-1r0ek5c-1 eJzVLt"
    name="search"
  />
</form>
`;
